src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import geometry.euclidean.basic
+import geometry.manifold.charted_space
+import analysis.normed_space.inner_product
+
+/-!
+# Conformal Maps on Inner Product Spaces
+
+A map between real inner product spaces `X`, `Y` is `conformal_at` some point `x` in `X` if it is
+real differentiable and its differential at that point is a nonzero constant multiple of
+a linear isometry equivalence.
+
+## Main results
+
+* `conformal_at`: the main definition
+* The conformality of the composition of two conformal maps, the identity map
+  and multiplications by a constant
+* `conformal_at_iff`: an equivalent definition of the conformality at some point
+* `conformal_at_preserves_angle`: if a map is `conformal_at` some point, then its differential
+                                  preserves all angles at that point
+* `conformal_groupoid`: the groupoid of conformal local homeomorphisms
+
+## Tags
+
+conformal
+
+## Warning
+
+The definition of conformality in this file does NOT require the maps to be orientation-preserving.
+Maps such as the conjugate function on the complex plane is considered as conformal maps.
+-/
+
+noncomputable theory
+
+section conformality
+
+/-- A continuous linear map `f'` is said to be conformal if it's
+a nonzero multiple of a bijective linear isometry. -/
+def is_conformal_map {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+  (f' : X →L[ℝ] Y) :=
+∃ (c : ℝ) (hc : c ≠ 0) (lie : X ≃ₗᵢ[ℝ] Y), (f' : X → Y) = (λ y, c • y) ∘ lie
```

Comment:
For someone with more mathematical background; Does it make sense to generalize this?
```suggestion
def is_conformal_map {R X Y : Type*}
  [semiring R] [topological_space X] [semi_normed_group X] [topological_space Y]
  [semi_normed_group Y] [module R X] [module R Y] (f' : X →L[R] Y) :=
∃ (c : R) (hc : c ≠ 0) (lie : X ≃ₗᵢ[R] Y), (f' : X → Y) = c • lie
```

(note that `c • lie` is a shorter spelling of `(λ y, c • y) ∘ lie` here, and not really relevant to the generalization)

Reply:
Will the RHS be automatically coerced into an `X → Y` function?

Reply:
Yes, you can `#print is_conformal_map` to check - it shows `⇑f' = c • ⇑lie` at the end of the term.

My question here is more about whether generalizing away from `ℝ` makes sense it - it at least seems that you can generalize a handful of the subsequent lemmas if you do.

Reply:
> Yes, you can `#print is_conformal_map` to check - it shows `⇑f' = c • ⇑lie` at the end of the term (even though the elaborator does not allow you to type that.
> 
> My question here is more about whether generalizing away from `ℝ` makes sense it - it at least seems that you can generalize a handful of the subsequent lemmas if you do.

Thank you. I think all the trivial lemmas could be generalized.

Reply:
See comment at
https://github.com/leanprover-community/mathlib/pull/8367#discussion_r672575837